### PR TITLE
[Validator] [Validation] Document the auto_mapping config option

### LINKED ISSUE #18245: [Validator][Doctrine] Add Note That Automatic Validation is Disabled by Default
The subsection in question is [Databases and the Doctrine ORM: Validating Objects](https://symfony.com/doc/current/doctrine.html#validating-objects).

Additionally, we may want to elaborate on how to enable this feature.

Ref: symfony/symfony-docs#12728, symfony/symfony-docs#11132, symfony/recipes#664

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #18245.

Please, review the XML and PHP config example because I just made them up.

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -2499,6 +2499,67 @@ enabled
...
+    .. code-block:: php
+
+        // config/packages/framework.php
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $framework): void {
+            $framework->validation()
+                ->autoMapping()
+                    ->paths([
+                        'App\Entity\' => [],
```
Comment (xabbuh) [inline reference/configuration/framework.rst]:
```suggestion
                        'App\Entity\\' => [],
```

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -2499,6 +2499,67 @@ enabled
...
+                        'Foo\' => ['Foo\Some\Entity', 'Foo\Another\Entity'],
```
Comment (xabbuh) [inline reference/configuration/framework.rst]:
```suggestion
                        'Foo\\' => ['Foo\Some\Entity', 'Foo\Another\Entity'],
```

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -2499,6 +2499,67 @@ enabled
...
+
+            <framework:config>
+                <framework:validation>
+                    <framework:auto-mapping>
+                        <framework:namespace>App\Entity\</framework:namespace>
+                        <framework:service/>
+
+                        <framework:namespace>Foo\</framework:namespace>
+                        <framework:service>Foo\Some\Entity</framework:service>
+                        <framework:service>Foo\Another\Entity</framework:service>
```
Comment (7ochem) [inline reference/configuration/framework.rst]: I'm not very experienced with config XML, but the namespace should be an attribute as it seems from https://github.com/symfony/symfony/blob/6c85287f18120b6a00014eec806a6427bbb6e080/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1030

Should it be like this?
```xml
<framework:service namespace="App\Entity\"/>
<framework:service namespace="Foo\">Foo\Some\Entity</framework:service>
<framework:service namespace="Foo\">Foo\Another\Entity</framework:service>
```

**Comment by javiereguiluz** [comment]
Thanks for your reviews! I've made all the suggested changes.

**Pull request closed without merging**